File: Clocktower.Server/Data/SessionSyncState.cs
using Clocktower.Server.Data.Dto;

namespace Clocktower.Server.Data;

public class SessionSyncState
{
    public required GameTime GameTime { get; init; }
    public required string Jwt { get; init; }
    public required DiscordTownDto? DiscordTown { get; init; }
    public required TimerState Timer { get; init; }
}

File: Clocktower-UI/src/types/sessionSyncState.ts
import {
    DiscordTown,
    type GameTime,
    NominationSession,
    Script,
    TalkRequest,
    type TimerState
} from "@/types";

export type SessionSyncState = {
    gameTime: GameTime,
    jwt: string,
    discordTown?: DiscordTown;
    timer?: TimerState;
    script?: Script;
    nominationSession?: NominationSession | null;
    talkRequests?: TalkRequest[] | null;
};